addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -701,6 +701,13 @@ def action_refuse(self):
         if any(holiday.state not in ['confirm', 'validate', 'validate1'] for holiday in self):
             raise UserError(_('Allocation request must be confirmed or validated in order to refuse it.'))
 
+        days_per_allocation = self.holiday_status_id._get_employees_days_per_allocation(self.employee_id.ids)
+
+        for allocation in self:
+            days_taken = days_per_allocation[allocation.employee_id.id][allocation.holiday_status_id][allocation]['virtual_leaves_taken']
+            if days_taken > 0:
```

Comment:
I guess it leaves out some edge cases in case of multiple allocations on that leave type (A and B both override the same period, A has a days_taken count, so we cannot delete it though be could be handling the leave)

Reply:
Summary of our last discussion: This is an edge case that should rarely occur so we shouldn't overengineer its handling as long as it handles all cases correctly (ie doesn't bring the db in an inconsistent state) by presenting the user with a clear error message that lets them decide the next course of action.

Concerning writing, I made a different task on which we can discuss further how we end up handling writes for confirmed allocations so that that issue doesn't have to slow down this task: https://www.odoo.com/web#cids=1&menu_id=4720&action=333&active_id=1251&model=project.task&view_type=form&id=3381984